Userland/Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -4089,8 +4089,9 @@ void Document::ensure_animation_timer()
                 m_animation_driver_timer->stop();
                 return;
             }
-
-            update_animations_and_send_events(window()->performance()->now());
+            auto* window_or_worker = dynamic_cast<HTML::WindowOrWorkerGlobalScopeMixin*>(&realm().global_object());
+            VERIFY(window_or_worker);
```

Comment:
I'm not a massive fan of this cast to be honest (although I've also used it elsewhere). I believe it is guaranteed to always succeed, but I'm not 100% sure. Is there a better way to do this?

Reply:
If this become a performance problem, we could do this via a virtual function or something instead, but this seems okay for now.